File: app/src/main/java/com/example/kaliumapp/data/database/entities/WaterIntake.kt
package com.example.kaliumapp.data.database.entities

import androidx.room.*

@Entity(
    tableName = "water_intake",
    foreignKeys = [ForeignKey(
        entity = DailyWaterSummary::class,
        parentColumns = ["summary_id"],
        childColumns = ["summary_id"],
        onDelete = ForeignKey.CASCADE
    )],
    indices = [Index(value = ["summary_id"])]
)
data class WaterIntake(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "intake_id")
    val intakeId: Int = 0,
    @ColumnInfo(name = "summary_id")
    val summaryId: Int,
    @ColumnInfo(name = "intake_ml")
    val intakeMl: Int,
    @ColumnInfo(name = "intake_time")
    val intakeTime: Long = System.currentTimeMillis()
)

File: models/water_intake.py
from sqlalchemy import Column, Integer, BigInteger, ForeignKey
from sqlalchemy.orm import relationship
from database.database import Base

class WaterIntake(Base):
    __tablename__ = "water_intake"

    intake_id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    summary_id = Column(Integer, ForeignKey("daily_water_summary.summary_id"), nullable=False)
    intake_ml = Column(Integer, nullable=False)
    intake_time = Column(BigInteger, nullable=False)
    is_synced = Column(Integer, default=0, nullable=False)

    user = relationship("User", back_populates="water_intakes")
    summary = relationship("DailyWaterSummary", back_populates="water_intakes")